### PR TITLE
Extensions V2, speedier to distribute than V1 thanks to Bref V2 decisions :zap: 

### LINKED ISSUE #375: Bref 2.0
Hey @Nyholm!

As I work on Bref v2, I started testing the impact on the extra extensions. I think it _should_ be minimal.

So far I've been able to build redis, decimal, and ds by just replacing:

```diff
- FROM bref/build-php-$PHP_VERSION:1.7.4 AS ext
+ FROM --platform=linux/amd64 bref/build-php-$PHP_VERSION:latest AS ext
```

Note: I used `latest` so that it uses the local image I built from https://github.com/brefphp/aws-lambda-layers Eventually you'll be able to use a real version like 2.0.0 or something similar.

The main change is to force using the x86 platform, else when building on M1 (for example) it tries to pick up the ARM images.

Bref v2 will also contain ARM images, I'm not sure if that implies also building extensions on ARM specifically, but I guess it might be the case. Not sure if this is something you want to tackle outright.

### PR DESCRIPTION
**Pull request by shouze**
Closes #375

# TODO

- [ ] AMQP
- [ ] Blackfire
- [x] Calendar
- [x] Cassandra
- [x] Decimal
- [x] DS
- [ ] Elastic APM
- [ ] GD
- [x] gnupg
- [x] GMP
- [x] gRPC
- [ ] ~Igbinary~ ping @mnapoli @Nyholm already included into redis & memcached
- [ ] Imagick
- [x] IMAP
- [x] LDAP
- [x] Mailparse
- [ ] MaxMind DB
- [x] Memcache
- [x] Memcached
- [x] MongoDB
- [ ] ~MsgPack~ ping @mnapoli @Nyholm already included into redis & memcached
- [ ] Newrelic
- [ ] ODBC Snowflake
- [ ] Oracle
- [x] Pcov
- [ ] PostgreSQL
- [x] RdKafka
- [x] Redis
- [ ] ~Redis-Igbinary~ ping @mnapoli @Nyholm already included into redis & memcached
- [ ] Scout APM
- [ ] Scrypt
- [ ] SPX
- [x] SSH2
- [ ] Symfony Runtime
- [ ] Microsoft SQLSRV
- [x] Tidy
- [x] UUID
- [x] Xdebug
- [ ] Xlswriter
- [x] Yaml

### CODE REVIEW

File: layers/redis/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-RUN pecl install --force redis
-RUN cp `php-config --extension-dir`/redis.so /tmp/redis.so
-RUN echo 'extension=/opt/bref-extra/redis.so' > /tmp/ext.ini
 
 # Build the final image with just the files we need
 FROM scratch
 
 # Copy things we installed to the final image
-COPY --from=ext /tmp/redis.so /opt/bref-extra/redis.so
+COPY --from=ext /usr/lib64/liblzf.so* /opt/lib/
```
Comment (mykiwi) [inline layers/redis/Dockerfile:19]: I think we should do `COPY --link ...`
cf https://youtu.be/mgW61S7izcc?t=886

(took from the video)
Screenshot 2022-12-11 at 15-33-29 What's New in Docker Build

Comment (mnapoli) [reply layers/redis/Dockerfile:19]: Wow that's so interesting! Will probably try to use this in https://github.com/brefphp/[aws-lambda-layers](https://github.com/brefphp/aws-lambda-layers) as well!

Comment (shouze) [reply layers/redis/Dockerfile]: @mykiwi I've added it https://github.com/brefphp/extra-php-extensions/pull/377/commits/d26f8dd31bd8b57d50c3a884c56da76c209e9803, is it what you had in mind?

Comment (mykiwi) [reply layers/redis/Dockerfile]: yes
I just have a little doubt about the `--link` between the `--from=something` & `source dest`, I would probably go for `COPY --link --from=something source dest` but if this does not break :shrug: 
we should probably do this trick on every COPY of the PR (not only for redis)

Comment (shouze) [reply layers/redis/Dockerfile]: I've tested both, same result. Sounds logic as they just are options so hopefully order has no importance.

:+1: for using it everywhere.

**Comment by shouze** [comment]
I think I've done most of the easy ones. I'm ok for some help to finish remaining ones as I won't have a lot of time to work on it during this week.

File: layers/gd/Dockerfile [hunk 2]
```diff
@@ -1,50 +1,16 @@
...
-RUN echo 'extension=/opt/bref-extra/gd.so' > /tmp/ext.ini
+RUN <<eot bash -e
+yum install -y php-gd
+echo 'extension=gd.so' > /tmp/ext-gd.ini
+eot
 
+# Build the final image with just the files we need
 FROM scratch
 
-COPY --from=ext /usr/lib64/libXpm.so.4 /opt/bref/lib64/libXpm.so.4
```
Comment (jon-nunan) [inline layers/gd/Dockerfile:40]: I think you'll need to keep most of these for gd to work properly.

Messing around on a local branch I had to add the following for it to load
```
COPY --link --from=ext /usr/lib64/libgd.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libXpm.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libX11.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libpng15.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libjpeg.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libfreetype.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libfribidi.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libglib-2.0.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libgraphite2.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libharfbuzz.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libjbig.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libraqm.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libtiff.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libxcb.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libXau.so* /opt/lib/
COPY --link --from=ext /usr/lib64/libwebp.so* /opt/lib/
COPY --link --from=ext /usr/lib64/php/modules/gd.so* /opt/bref/extensions/
COPY --link --from=ext /tmp/*.ini /opt/bref/etc/php/conf.d/
```

**Comment by shouze** [comment]
closing it since we're back to compiled php in latests 2.0 Beta

**Pull request closed without merging**